lib/beanstream/payments_api.rb

```diff
@@ -129,9 +129,12 @@ def make_payment(payment)
       val = transaction_post("POST", make_payment_url, Beanstream.merchant_id, Beanstream.payments_api_key, payment)
     end
     
-    def complete_preauth(transaciton_id, amount)
+    def complete_preauth(transaciton_id, amount, order_number)
```

Comment:
You could make the last param (order_number) optional and only included if provided.